### DISCUSSION TITLE
Is Bazecor able to determine which OS you're using, similar to QMK?

### DISCUSSION THREAD

**Discussion by jasonminsookim** [Dygmalab/Bazecor#398] [Q&A]
I work with a mac and windows computer for programming work. I'd love to be able to use the same key for both control/command keys dependent on which OS I'm currently using.

**Comment by javierguzman** [comment]
I have tried with my laptops and the mac cmd key on Windows is the Windows key which I think it makes sense. I use control on both systems.

**Reply by jasonminsookim** [reply]
I want to use command/windows c and command/windows v for copy paste, so that I don't have to context switch between OS. Is this possible? @javierguzman

**Reply by javierguzman** [reply]
Right now is not possible as they are literally two different keys (Cmd for copy/paste on macOS and Ctrl for copy/paste on Windows). I do not know enough about keyboards to figure out an easy way to do that, however, a possible solution would be to detect automatically the OS and then use one layer or another depending on the one you are using, but I would say is not a clean one as we would need more memory to store more layers, etc. In any case, right now people is busy with the new Defy which should be released soon, so after that,  we will continue adding features so we will revisit what people want the most.

**Comment by GazHank** [comment]
There is a [pluggin available for the underlying firmware](https://github.com/keyboardio/Kaleidoscope/tree/master/plugins/Kaleidoscope-HostOS) to detect the host OS and run some OS specific code; but I don't believe it is incorporated into the current raise firmware; nor is any enabling configuration available within Bazecor.

It should be reasonable simple to add the plugin to the firmware, and then perhaps have either each os default to a different layer, or swap some of the key mappings (I think a default layer per OS would be simplest?)
